### PR TITLE
[libffi] Fix processors

### LINKED ISSUE #16269: [libffi] build failure on macOS
**Host Environment**
- OS: macOS 11 with Apple Silicon
- Compiler: `Apple clang version 12.0.0 (clang-1200.0.32.29)`

**To Reproduce**
`./vcpkg install libffi`

**Failure logs** 
```
Starting package 83/89: libffi:x64-osx
Building package libffi[core]:x64-osx...
-- Downloading https://github.com/libffi/libffi/archive/v3.3.tar.gz -> libffi-libffi-v3.3.tar.gz...
-- Extracting source /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/libffi-libffi-v3.3.tar.gz
-- Applying patch win64-disable-stackframe-check.patch
-- Applying patch win32-disable-stackframe-check.patch
-- Using source at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/libffi/src/v3.3-8557367958.clean
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:108 (message):
    Command failed: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/bin/cmake /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/libffi/src/v3.3-8557367958.clean -DFFI_CONFIG_FILE=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/ports/libffi/fficonfig.h -DCMAKE_MAKE_PROGRAM=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/tools/ninja-1.10.1-osx/ninja -DCMAKE_SYSTEM_NAME=Darwin -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/scripts/toolchains/osx.cmake -DVCPKG_TARGET_TRIPLET=x64-osx -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg -D_VCPKG_INSTALLED_DIR=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -DCMAKE_OSX_ARCHITECTURES=x86_64 -DFFI_SKIP_HEADERS=ON -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/libffi_x64-osx/debug
    Working Directory: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/libffi/x64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/libffi/config-x64-osx-dbg-out.log
      /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/libffi/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:328 (vcpkg_execute_required_process)
  ports/libffi/portfile.cmake:17 (vcpkg_configure_cmake)
  scripts/ports.cmake:128 (include)


Error: Building package libffi:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libffi:x64-osx
  Vcpkg version: 2021-01-13-unknownhash
```
[CMakeOutput.log](https://github.com/microsoft/vcpkg/files/5998413/CMakeOutput.log)
[CMakeError.log](https://github.com/microsoft/vcpkg/files/5998415/CMakeError.log)
[config-x64-osx-dbg-err.log](https://github.com/microsoft/vcpkg/files/5998418/config-x64-osx-dbg-err.log)
[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/5998419/config-x64-osx-dbg-out.log)

### PR DESCRIPTION
**Pull request by autoantwort**
**Describe the pull request**

- What does your PR fix? Fixes #16269

### CODE REVIEW

File: ports/libffi/CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ endif()
 # config variables for ffi.h.in
 set(VERSION 3.3)
 
-set(KNOWN_PROCESSORS x86 x86_64 AMD64 ARM ARM64 i386 armv7l armv7-a aarch64)
+set(KNOWN_PROCESSORS x86 x86_64 AMD64 ARM arm64 i386 armv7l armv7-a aarch64)
```
Comment (ras0219-msft) [inline ports/libffi/CMakeLists.txt:14]: I think this is doing the wrong thing and falls into the `elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin"  AND CMAKE_SIZEOF_VOID_P EQUAL 4)` case (`X86_DARWIN`).

I think a better approach here would be to use `TO_LOWER` on `CMAKE_SYSTEM_PROCESSOR` and lowercase all the names.

```cmake
string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" lower_system_proc)
```

Comment (ras0219-msft) [reply ports/libffi/CMakeLists.txt:14]: Then, the `MATCHES "ARM"` below changes to `MATCHES "arm"`

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/libffi/CMakeLists.txt (hunk 1 above)
Comment (autoantwort) [reply ports/libffi/CMakeLists.txt]: Ok done.

**Review by NancyLi1013 — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]
I did a bit more rearranging to better account for previous behavior.

**Comment by autoantwort** [comment]
@ras0219-msft you approved the changes, but merge is still needed

**Comment by ras0219-msft** [comment]
If the CI is still running, I approve but can't merge yet :)

**Pull request merged**